src/components/Form/СreateUserAccountForm/CreateUserAccountForm.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useMemo, useState } from 'react';
+import { Box, TextField, Typography } from '@material-ui/core';
+import i18n from 'i18next';
+import { langTokens } from '../../../locales/localizationInit';
+import { UserAccountButton } from '../Buttons';
+import { useStyles } from './styles/CreateUserAccountForm.style';
+import { regex } from '../../../views/Profile/PersonalInfo/constants/regex';
+import { UserEmailType } from '../../../views/Profile/PersonalInfo/types';
+
+type CreateUserAccountFormPropsType = {
+  usersEmails: UserEmailType;
+  onClick: (
+    e: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+    email: string,
+  ) => void;
+  currentUserEmail?: string;
+  isLoading?: boolean;
+};
+type ErrorFieldsType = {
+  email: string;
+};
+type VisitedFieldsType = {
+  email: boolean;
+};
+export const CreateUserAccountForm: React.FC<CreateUserAccountFormPropsType> = ({
+  onClick,
+  usersEmails,
+  currentUserEmail,
+  isLoading,
+}) => {
+  const [userEmail, setUserEmail] = useState('');
```

Comment:
what if we use `react-hook-form` here? would code be less verbose?

Reply:
I think for now it`s more simple to leave it as it is